### ISSUE TITLE
How to install docker-compose v2 on Linux

### ISSUE THREAD

**Issue by rfay** [docker/compose-cli#2012]
**Description**

I'm trying to allow DDEV to use docker-compose v2 and don't have the resources for a whole set of tests on every macOS variant... but I thought it would be nice to run all the tests using docker-compose v2 on Linux (where it hasn't been introduced yet to my knowledge?)

I followed the linux instructions in https://github.com/docker/compose-cli/blob/main/INSTALL.md but that doesn't do anything I can detect (`curl -L https://raw.githubusercontent.com/docker/compose-cli/main/scripts/install/install_linux.sh | sh`) (and doesn't enable the `docker compose` command either).

How can I use docker-compose v2 on Linux?

**Comment by ibnesayeed** [comment]
As per https://docs.docker.com/compose/cli-command/#install-on-linux

```
$ mkdir -p ~/.docker/cli-plugins/
$ curl -SL https://github.com/docker/compose-cli/releases/download/v2.0.0-rc.1/docker-compose-linux-amd64 -o ~/.docker/cli-plugins/docker-compose
$ chmod +x ~/.docker/cli-plugins/docker-compose
```

Note: The version number may change, obviously!

**Comment by rfay** [comment]
Hmm, after taking those actions on Ubuntu 20.04 I see:
```bash
docker-compose --version
docker-compose version 1.28.5, build c4eb3a1f
```

I do see that `docker compose` then becomes a thing, but docker-compose is unaffected AFAICT.

Also, would be good to maintain the [INSTALL.md](https://github.com/docker/compose-cli/blob/main/INSTALL.md)

**Comment by ibnesayeed** [comment]
> ... but docker-compose is unaffected AFAICT.

Yes, that's a separate took.

**Comment by rfay** [comment]
The question here is "How to install docker-compose v2 on Linux"

Hundreds or thousands of tools wrap docker-compose. An installation technique to make compose-cli be "docker-compose" is essential for those tools.

**Comment by mat007** [comment]
@ndeloof yes but then you can’t use ` docker-compose `  anymore, you need to type ` docker compose ` .
I think what @rfay is asking for is a linux version of the switch/shim that Desktop ships.

**Comment by ndeloof** [comment]
@mat007 makes sense. Unfortunately this codebase has been designed to support canary testing, but we could extract the core redirection logic into a public component as part of compose.v2 codebase

**Comment by rfay** [comment]
The shortest-term view is I'd like to be able to run all ddev tests against compose v2 on Linux... because github actions is there and free. That's just to see if we can finally get compose v2 compatible and not have to embargo it.

But the long term view is... surely you don't want Linux users of docker-compose to stay on v1 because you've never provided a migration path? It seems like docker-compose on linux deserves first-class status. And yes, I know that supposedly everybody can just start using `docker compose` IF you package that, which you're not yet if I understand. But since every user of docker-compose has usages with docker-compose wired into it, that migration may never take place and all linux will stay on v1.

**Comment by ndeloof** [comment]
the point is that we can't bundle `compose` CLI plugin in `docker` Linux package until it get released as G.A.
As "installing" is just about donwloading adequate binary under $HOME/.docker/cli-plugins, I don't consider this a blocker for Linux users...

**Comment by thaJeztah** [comment]
> the point is that we can't bundle compose CLI plugin in docker Linux package until it get released as G.A.

I think it'd be ok to publish a package (before GA); https://github.com/docker/docker-ce-packaging/pull/553 implements it as a separate package (so users will be able to "opt-in" by installing the package).

For the transition `docker-compose` > `docker compose`, perhaps a transitional "compatibility" package (https://github.com/docker/compose-cli/issues/2012#issuecomment-906402612) that provides the alias/redirect could be an option to ease the migration.
